WorldState.Extensions/WorldState.Extensions.csproj

```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
```

Comment:
We'll see if we upgrade the .NET Standard version to 2.0, I wouldn't be against it, but as per my comment on #16, I'd rather keep the version to an older one for compatibility's sake.

Reply:
This is an extension project. I think it should always target the latest Standard and add functionalities that are only available in later standards (like I mentioned returning _System.Drawing.Color_).

In production this would be compiled into a separate package. It will have no impact on the original project's target framework.

If at the end there wasn't really much code here, it would mean that .NET Standard 1 was a fitting choice, and you could simply get rid of these extensions.

Reply:
It's possible to target multiple frameworks as well. You can have compile-time constants defined for each framework version if you wanted and have features implemented differently depending on which framework is being targeted:

```xml
<TargetFrameworks>netstandard1.6;netstandard2.0;netstandard2.1;etc</TargetFrameworks>
```

... and later
```xml
<ItemGroup Condition="'$(TargetFramework)' == 'netstandard1.6'">
    <!-- ... -->
</ItemGroup>
```

Reply:
Thank your for the input. The downside for multitargeting is having to do many `#if`s and `#endif`s, because some code for the latest runtime wouldn’t be recognized by the older compiler. Other than that multitargeting is indeed a good alternative to extension packages.

.NET Standard 2.1 also added global nullable reference types switch (reference types default to non-null like value types unless explicitly marked so) which is IMO a great feature to have.

Reply:
I agree that the global nullable reference type switch is a great feature, however it does have some flaws regarding generic types. Additionally, it is still recommended to provide your own null-checking at the start of your methods even for non-nullable reference types because the nullability checking is implemented using attributes and code analysis rather than actual runtime null checking.

Regardless, it's still possible to use features from `netstandard2.1` in older versions of the framework by providing the implementations yourself (or relying on a third party library to do so). This means a project could still use c# 8 features while targeting a framework as early as `netstandard1.0`.

Also, I don't think there would be *that* many `#if`s and `#endif`s using multitargeting. What particular features are you looking for in `netstandard2.0`/`netstandard2.1`?

Reply:
I’ve mentioned `System.Drawing.Color` previously in this thread so that’s definitely one of them. I can’t think of any other right now, sorry, wasn’t active here for a while.

Anyway, agreed, not `that` many conditionals. Just pointing that out potential drawback for the whole picture.

I personally prefer extensions as they (1) are optionally, never hurt; (2) keep base project/package behavioral consistent across standards/platforms, you’d never have to change existing code base whatsoever.